### PR TITLE
feat: guardian update and forge unit tests

### PR DESCRIPTION
**Pull request by jtlacci**

### CODE REVIEW

File: test/manageMembers.spec.js [hunk 1]
```diff
@@ -222,4 +222,84 @@ describe("manage members integration test", () => {
       expect(await ethersSafe.isOwner(charlie.address)).to.eq(true);
     });
   });
+
+  describe("when managing signers as safe", () => {
+    let { signers, memberToken, ethersSafe, createPodHelper, safeExecutionHelper } = {};
+    let [, , , charlie, dave] = [];
```
Comment (0xGrapefruit) [inline test/manageMembers.spec.js:228]: what is this lol

File: test/foundry/MemberTeller.t.sol [hunk 2]
```diff
@@ -0,0 +1,57 @@
+pragma solidity ^0.8.7;
+
+import "forge-std/Test.sol";
+import "../mocks/MockMemberToken.sol";
+import "../../contracts/MemberTeller.sol";
+
+contract MemberTellerInternalTest is Test, MemberTeller {
```
Comment (0xGrapefruit) [inline test/foundry/MemberTeller.t.sol:7]: Aren't these tests duplicated in js? Should we keep one or the other?

File: contracts/ControllerV1.sol [hunk 3]
```diff
@@ -403,13 +442,19 @@ contract ControllerV1 is IControllerV1, SafeTeller, Ownable {
...
-            uint8(data[0]) == CREATE_EVENT
-        ) return;
+        if (operator == address(this)) {
+            // if create or sync event than side effects have been pre-handled
+            if (data.length > 0) {
+                if (uint8(data[0]) == CREATE_EVENT) return;
+                if (uint8(data[0]) == SYNC_EVENT) return;
+            }
+            // because 1155 burn doesn't allow data we use a burn sync flag to skip side effects
+            if (BURN_SYNC_FLAG == true) {
```
Comment (0xGrapefruit) [inline contracts/ControllerV1.sol:453]: neat pattern

File: contracts/MemberTeller.sol [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    bytes4 public constant ENCODED_SIG_SWAP_OWNER =
+        bytes4(keccak256("swapOwner(address,address,address)"));
+
+    uint8 internal constant SYNC_EVENT = 0x02;
+
+    constructor(address _memberToken) {
+        memberToken = IMemberToken(_memberToken);
+    }
+
+    function getSyncData() internal pure returns (bytes memory) {
```
Comment (0xGrapefruit) [inline contracts/MemberTeller.sol:20]: What is sync data?

**Review by 0xGrapefruit — APPROVED** [review]

File: test/manageMembers.spec.js (hunk 1 above)
Comment (jtlacci) [reply test/manageMembers.spec.js:228]: was just destructuring hack to make copy paste easier

File: test/foundry/MemberTeller.t.sol (hunk 2 above)
Comment (jtlacci) [reply test/foundry/MemberTeller.t.sol:7]: I deleted most of the js unit tests, the remaining js tests are integration and test state updates of flows across multiple contracts

File: contracts/MemberTeller.sol (hunk 4 above)
Comment (jtlacci) [reply contracts/MemberTeller.sol:20]: it creates the sync flag so Controller knows to skip side effects

Comment (0xGrapefruit) [reply contracts/MemberTeller.sol:20]: why is it called the sync flag? did you make that up or is that from something else?

**Pull request merged**